mllib/src/main/scala/org/apache/spark/mllib/linalg/Vectors.scala

```diff
@@ -63,20 +63,27 @@ sealed trait Vector extends Serializable {
             Vectors.equals(0 until d1.size, d1.values, s1.indices, s1.values)
           case (_, _) => util.Arrays.equals(this.toArray, v2.toArray)
         }
-      }
       case _ => false
     }
   }
 
+  /**
+   * Returns a hash code value for the vector. The hash code is based on its size and its nonzeros
+   * in the first 16 entries, using a hash algorithm similar to [[java.util.Arrays.hashCode]].
+   */
   override def hashCode(): Int = {
```

Comment:
Why is Vector.hashCode overridden, even though it will never be called (since Sparse & Dense Vector both override it)?